conn.go

```diff
@@ -32,6 +32,10 @@ var (
 	ErrSSLNotSupported           = errors.New("pq: SSL is not enabled on the server")
 	ErrSSLKeyHasWorldPermissions = errors.New("pq: Private key file has group or world access. Permissions should be u=rw (0600) or less.")
 	ErrCouldNotDetectUsername    = errors.New("pq: Could not detect default username. Please provide one explicitly.")
+
+	errUnexpectedReady = errors.New("unexpected ReadyForQuery")
+	errNoRowsAffected  = errors.New("no RowsAffected available after the empty statement")
+	errNoLastInsertId  = errors.New("no LastInsertId available after the empty statement")
```

Comment:
I don't expect them to be used in that way. The errors are named here for the tests. I do not think they need to be exported because these errors are very rare (as evidenced by the fact that it was previously a panic if they occurred and that went undetected until now). It is incredibly unlikely that another library cares if an empty string is the cause of an error instead of just caring that there was an error at all.

However if you disagree I'm happy to export them. But I think that adds some unused clutter to the exported API that doesn't need to be present.